exporters/prometheus/src/integrationTest/java/io/opentelemetry/exporter/prometheus/PrometheusCollectorIntegrationTest.java

```diff
@@ -0,0 +1,187 @@
+  private static SdkMeterProvider meterProvider;
+  private static OtlpGrpcServer grpcServer;
+  private static GenericContainer<?> collector;
+
+  @BeforeAll
+  static void beforeAll() {
+    PrometheusHttpServer prometheusHttpServer = PrometheusHttpServer.builder().setPort(0).build();
+    prometheusPort = prometheusHttpServer.getAddress().getPort();
+    meterProvider = SdkMeterProvider.builder().registerMetricReader(prometheusHttpServer).build();
+    exposeHostPorts(prometheusPort);
+
+    grpcServer = new OtlpGrpcServer();
+    grpcServer.start();
+    exposeHostPorts(grpcServer.httpPort());
+
+    collector =
+        new GenericContainer<>(DockerImageName.parse(COLLECTOR_IMAGE))
+            .withEnv("APP_ENDPOINT", "host.testcontainers.internal:" + prometheusPort)
+            .withEnv("LOGGING_EXPORTER_LOG_LEVEL", "INFO")
+            .withEnv(
+                "OTLP_EXPORTER_ENDPOINT", "host.testcontainers.internal:" + grpcServer.httpPort())
+            .withClasspathResourceMapping(
+                "otel-config.yaml", "/otel-config.yaml", BindMode.READ_ONLY)
+            .withCommand("--config", "/otel-config.yaml")
+            .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("otel-collector")))
+            .withExposedPorts(COLLECTOR_HEALTH_CHECK_PORT)
+            .waitingFor(Wait.forHttp("/").forPort(COLLECTOR_HEALTH_CHECK_PORT));
+    collector.start();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    meterProvider.shutdown().join(10, TimeUnit.SECONDS);
+    grpcServer.stop().join();
+    collector.stop();
+  }
+
+  @AfterEach
+  void afterEach() {
+    grpcServer.reset();
+  }
+
+  @Test
+  void endToEnd() {
+    Meter meter = meterProvider.meterBuilder("test").build();
+
+    meter
+        .counterBuilder("requests")
+        .build()
+        .add(3, Attributes.builder().put("animal", "bear").build());
+
+    await()
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(() -> assertThat(grpcServer.metricRequests.size()).isGreaterThan(0));
+
+    ExportMetricsServiceRequest request = grpcServer.metricRequests.get(0);
+    Assertions.assertThat(request.getResourceMetricsCount()).isEqualTo(1);
+
+    ResourceMetrics resourceMetrics = request.getResourceMetrics(0);
+    // TODO: when otel-collector-contrib:0.55.0 is released, update to include resource attributes
+    // which it will scrape from target info and include on the OTLP resource
+    assertThat(resourceMetrics.getResource().getAttributesList())
+        .containsExactlyInAnyOrder(
+            stringKeyValue(ResourceAttributes.SERVICE_NAME.getKey(), "app"),
+            stringKeyValue(ResourceAttributes.HOST_NAME.getKey(), "host.testcontainers.internal"),
+            stringKeyValue("job", "app"),
+            stringKeyValue("instance", "host.testcontainers.internal:" + prometheusPort),
+            stringKeyValue("port", String.valueOf(prometheusPort)),
+            stringKeyValue("scheme", "http"));
```

Comment:
Here we see the resource attributes the collector prometheus receiver adds:
- It sets `service.name` from the name of the target
- Notably, it doesn't yet set `service.instance.id`, `net.host.name`, `net.host.port`, or `http.scheme` as described in the [spec](https://github.com/open-telemetry/opentelemetry-specification/blame/main/specification/metrics/data-model.md#L1291-L1300). 
- As noted in the TODO, the target info labels will appear as resource attributes in the `0.55.0` of the collector. Here's the [PR](https://github.com/open-telemetry/opentelemetry-specification/blame/main/specification/metrics/data-model.md#L1291-L1300) that adds that behavior. I tested it locally for now.

Reply:
Left a [comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9967#issuecomment-1170532799) on a relevant issue in the collector seeking to understand why the behavior is different from the spec.